lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -446,7 +446,7 @@ public void OnNext(ICompletedTask completedTask)
                 switch (_systemState.CurrentState)
                 {
                     case SystemState.TasksRunning:
-                        _taskManager.RecordCompletedTask(completedTask);
+                        _taskManager.RecordCompletedRunningTask(completedTask);
                         if (_taskManager.AreAllTasksCompleted())
```

Comment:
I would prefer to let all the tasks to complete by themselves in successful case so that everything is completed properly, even master task is completed earlier than others. 
For exceptional case, driver receives FailedEvalautor or FailedTask after master task is completed, this is rared case. In case it happens, we want just grab the result and shut down the system, because retry doesn't make sense any more in this scenario as entire iterations has been completed.